src/processor/operator/gds_call_shared_state.cpp

```diff
@@ -5,6 +5,17 @@ using namespace kuzu::common;
 namespace kuzu {
 namespace processor {
 
+offset_t NodeOffsetMaskMap::getNumMaskedNode() const {
+    if (!enabled_) {
```

Comment:
This looks a bit prone to bugs to me because you return 0 for two different reasons. One is if the NodeMask is not enabled second is if it's enabled but no node is marked in the mask. As a general principle outputs of functions should have one interpretation and lack of indicates a problem with the design. I would try to give clear meaning to each possible output of this function, e.g., return MAX_OFFSET or something or just throw an exception if this function should not be called if the mask was disabled.

Also, if for some reason you don't need to differentiate between the above 2 cases, will the code from 12-16 just return 0 if the mask was disabled? If so, you might consider removing the first if clause.

Reply:
Done

Reply:
Remove this check